versions/3.0.md

```diff
@@ -1,5 +1,133 @@
+
+```yaml
+hosts:
+  - scheme: http
+    host: foo.com
+    port: 8080
+    basePath: /bar
+  - scheme: https
+    host: foo.com
+    port: 8443
+    basePath: /baz
+```
+
+And remove the separate `host`, `basePath`, `schemes`.  Because the port will never be the same between http, https, it makes no sense to support multiple ones
+
+Fixes #489, #214
+
+
+## Why don't host and basePath support path templating? #169
+
+- [ ] Proposal: this belongs under parent #574
+
+## Support an operation to have multiple specifications per path (e.g. multiple POST operation per path) #182
+
+- [ ] Proposal: this is a payload issue, move to parent #586
+
+## Allow for overloading/aliasing of route path #213
+
+- [ ] Proposal: change in #70 would address this
+
+## Allow references between operations inside descriptions. #239
+
+- [ ] Proposal: this is really about _links_ and belongs in the #586 issue
+
+## Swagger.next: Let's not use hashes anymore #208
+
+- [ ] Proposal: while true, moving to arrays causes more problems than it solves.  Suggest we close this with no change.
+
+## Group multiple parameter definitions for better maintainability #445
+
+- [ ] Proposal: close.  Using global parameters will help with this, and references to arrays of parameters is just wrong.a
+
+
+## Allow documenting HATEOAS APIs (pathless operations) #577
+
+- [ ] Proposal: do not support pathless operations.  It's against the spirit of the specification as being declarative.  The support for links _inside_ a path though, will be covered by #574 and #586.  Remove reference to #560 and link to others.
+
+## Message format 'wrappers' #555
+
+- [ ] Proposal: Move discussion to parent issue #586, this is really about a payload definition, which is covered by that meta issue.  Non-structural.
+
+## Adding optional SLA definitions with the Spec #541
+
+- [ ] proposal: Add an optional SLA to the specification, follow the same locations / hierarchy as consumes/produces.  Need a SME to take this on and propose a structure.
+
+## No provision to group API's and then $ref it to reference it in other file #508
+
+- [ ] Proposal: I believe what this issue is about is to essentially include an entire path definition into a `path`.  This isn't possible currently because PIOs are in a hash, and you can not have a $ref in it. Here is a proposed solution:
+
+```yaml
+paths:
+  /pet:
+    get:
+      ...
+  /health:
+    $ref: '#/x-paths/health'
+  '#/x-paths': {}
+
+# note the location `x-paths` is just illustrative, it could be anywhere allowed by the schema
+x-paths:
+  /store:
+    get:
+      ...
+  /users:
+    get:
+      ...
+```
+
+treat any fragment or file reference as a "merge" against the root paths object.  It resolves like this:
+
+```yaml
+paths:
+  /pet:
+    get:
+      ...
+  /health:
+    $ref: '#/x-paths/health'
+  /store:
+    get:
+      ...
+  /users:
+    get:
+      ...
+```
+
```

Comment:
This requires further discussion.

We might as well just allow $ref’s, and multiple of them. We already break the $ref today with the Operation Object (and there’s currently no issue to change that). I’m generally in favor of breaking from $ref’s restrictions, and it makes for a much cleaner solution. If the concern is about $refs disappearing for being multiple keys, we can also have a single $ref with an array of strings. We can even call it something else like `references` to not be confused with `$ref`, and that would be rendered by the tools as multiple `$ref` occurrences. 

Using JSON Pointers as keys is odd, and it’s actually not a JSON Pointer if I understand correctly the usage, it’s the value of a JSON Reference which is not the same. Seems hack-ish and would be a pain to document in the spec.